clients/githubrepo/branches.go

```diff
@@ -481,8 +482,9 @@ nextRule:
 			}
 		}
 
+		targetsBranches := rule.Target != nil && *rule.Target == "branch"
 		for _, cond := range rule.Conditions.RefName.Include {
-			if cond == ruleConditionAllBranches {
+			if cond == ruleConditionAll && targetsBranches {
```

Comment:
Just to clarify, the issue was only a problem with tag protection rules which apply to all tags?

This is what I see with the graphql explorer for your repository:
```json
{
            "node": {
              "name": "Immutable tags with maintainer bypass",
              "enforcement": "ACTIVE",
              "target": "TAG",
              "conditions": {
                "refName": {
                  "exclude": [],
                  "include": [
                    "~ALL"
                  ]
                }
              },
```

I'm envisioning other cases where the tag include pattern would also match a branch name. So your approach in https://github.com/ossf/scorecard/pull/4699/commits/72db714db12c9d7d8dfa6d77106739ececc63622 may have been better?

Reply:
I'm good with the original approach, reverted to that